hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java

```diff
@@ -153,29 +153,29 @@ private void enrichHostHeader(Event event) {
 
   private Optional<String> getSanitizedHostValue(String value) {
     if (StringUtils.isNotBlank(value)) {
-      String host = sanitizeHostValue(value);
-      if (!LOCALHOST.equalsIgnoreCase(host)) {
-        return Optional.of(host);
+      Optional<String> host = sanitizeHostValue(value);
+      if (host.isPresent() && !LOCALHOST.equalsIgnoreCase(host.get())) {
+        return host;
       }
     }
     return Optional.empty();
   }
 
-  private String sanitizeHostValue(String host) {
+  private Optional<String> sanitizeHostValue(String host) {
     if (host.contains(COLON) && !host.startsWith(COLON)) {
-      return COLON_SPLITTER.splitToList(host).get(0);
+      return Optional.ofNullable(COLON_SPLITTER.splitToList(host).get(0));
```

Comment:
Here `host` is coming as `null` or as an empty string `""`, will that handle an empty string case?

Reply:
If `host` is empty, it shouldn't even match `host.contains(COLON)` right?